service_container/factories.rst

```diff
@@ -188,7 +188,7 @@ example takes the ``templating`` service as an argument:
             # ...
 
             AppBundle\Email\NewsletterManager:
-                factory:   'AppBundle\Email\NewsletterManagerFactory:createNewsletterManager'
+                factory:   'AppBundle\Email\NewsletterManagerFactory::createNewsletterManager'
```

Comment:
This change looks wrong to me. Using two colons means that method is treated as being static. But in here we do want to refer to a service with the id `AppBundle\Email\NewsletterManagerFactory`.

Reply:
Yes, for sure, I did not pay attention to `App\Email\NewsletterManagerFactory: ~`, it just looks like the full name of the class, which is confusing. It is necessary to return back the use of only one colon.
Sorry for the confusion :(

Reply:
No worries :) But I am going to close here then.

Reply:
Sure, but do not forget to return to use only one colon.

Reply:
I don't think there is anything to change, but a misunderstanding of what the `Definition` class does at that point. The thing is that your YAML config will already have been processed by the `YamlFileLoader` so that the `$factory` property in the `Definition` class will not be a  string anyway.